lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -81,6 +214,121 @@
   return std::u16string(u"uppered");
 }
 
+// Implementation of
+// https://tc39.es/ecma402/#sec-getoption
+// Funcion split into two depending on which data type is needed
+vm::CallResult<Option> getOptionBool(
+    const Options &options,
+    const std::u16string property,
+    const std::u16string type,
+    const std::vector<std::u16string> values,
+    const Option &fallback) {
+  // 1. Assert type(options) is object
+  // 2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  // 3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  // 4. Assert: type is "boolean" or "string".
+  // 5. If type is "boolean", then
+  if (type == u"boolean") {
+    if(!value->second.isBool()) {
+      // TODO: in what layer do we call into runtime to throw exceptions
+      //return runtime->raiseRangeError(u"Boolean option expected but not found");
+      return vm::ExecutionStatus::EXCEPTION;
```

Comment:
If the spec says you should throw at a certain point, then in general, that's where you should call into runtime. Let's avoid diverging from the spec wherever possible.